homeworks/03-data-structure/src/BinaryTree.ts

```diff
@@ -0,0 +1,105 @@
+import {BinaryTree, TreeNode, TraverseType} from "./BinaryTreeInterfaces"
+
+export class BinaryTreeClass<T> implements BinaryTree<T> {

```

Comment:
Here you implemented `BinaryTree` interface correctly. 

The `Class` suffix is redundant in the class name `BinaryTreeClass`. If you have name collisions, you can rename import or name the interface like `IBinaryTree`.